### PR TITLE
Rule S4069: Operator overloads should have named alternatives

### LINKED ISSUE #508: Rule S4069: Operator overloads should have named alternatives
[RSPEC-4069](https://jira.sonarsource.com/browse/RSPEC-4069)

Operator overloading is convenient but unfortunately not portable across languages. To be able to access the same functionality from another language you need to provide an alternate named method following the convention.

Contributes to [MMF-963](https://jira.sonarsource.com/browse/MMF-963).

### PR DESCRIPTION
**Pull request by valhristov**
Fix #508

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/OperatorOverloadsShouldHaveNamedAlternatives.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                }
+            },
+            SyntaxKind.OperatorDeclaration);
+        }
+
+        private static bool HasAlternativeMethod(IMethodSymbol operatorMethod, string operatorAlternativeMethodName)
+        {
+            return operatorAlternativeMethodName == null ||
+                operatorMethod.ContainingType
+                    .GetMembers(operatorAlternativeMethodName)
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/OperatorOverloadsShouldHaveNamedAlternatives.cs]: You are only checking for the method name, is it a good enough approximation? (Should you not check return type + parameter type and count?)

Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/OperatorOverloadsShouldHaveNamedAlternatives.cs]: Initially I checked for parameters and return type, but then I saw FxCop does not check that and simplified the code...

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/OperatorOverloadsShouldHaveNamedAlternatives.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+namespace Tests.Diagnostics
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/OperatorOverloadsShouldHaveNamedAlternatives.cs]: The rule does not enforce that every operator has a method. For example, if class has unary and binary operator+, having only one Add method would satisfy the rule. As Amaury mentioned, we are also not checking parameters.

I think these limitations are fair, but let's add tests that document that (ideally with short comments).

Comment (valhristov) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/OperatorOverloadsShouldHaveNamedAlternatives.cs]: Added a comment, test already exists (to some extent)

**Pull request merged**